activerecord/CHANGELOG.md

```diff
@@ -1,3 +1,11 @@
+*   Allow bypassing primary key/constraint addition in `implicit_order_column`
+
+    When specifying multiple columns in an array for `implicit_order_column`, adding
+    `nil` as the last element will prevent appending the primary key to order
+    conditions, thus improving performance for certain database queries.
```

Comment:
```suggestion
    conditions, allowing more precise control over which indexes may be used by
    generated queries, by accepting the risk of API misbehavior if the specified
    columns are not fully unique.
```

(Not a firm specific suggestion, just a quick take at trying to concisely provide some detail on the "performance" point, while also clarifying why it's not a "go fast" mode that everyone should always use.)